atc/container_limits_marshaller.go

```diff
@@ -23,20 +30,94 @@ func (c *ContainerLimits) UnmarshalJSON(limit []byte) error {
 	return nil
 }
 
-func (c *ContainerLimits) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	var data interface{}
+func ParseContainerLimits(data interface{}) (ContainerLimits, error) {
+	mapData, ok := data.(map[string]interface{})
+	if !ok {
+		mapData = make(map[string]interface{})
+	}
 
-	err := unmarshal(&data)
-	if err != nil {
-		return err
+	var c ContainerLimits
+
+	var memoryBytes uint64
+	var uVal int
+	var err error
+
+	// the json unmarshaller returns numbers as float64 while yaml returns int
+	for key, val := range mapData {
+		if key == "memory" {
```

Comment:
Can't you just look these up directly in the map?

Reply:
For posterity: probably, though this is actually just copied over from the decode hook; didn't write it myself.